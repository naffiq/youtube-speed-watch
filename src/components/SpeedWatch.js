import React from 'react';
import Time from './../utils/Time';

export default ({ seconds }) => {
  if (!seconds) {
    return (
      <div className="alert alert-warning">
        Enter video url
      </div>
    );
  }

  return (
    <div>
      <h2>Time to watch a video at different speeds:</h2>
      <table className="table">
        <tbody>
          <tr>
            <td>
              <span className="label label-default">x1</span>
            </td>
            <td> {Time.speedWatch(seconds)} </td>
          </tr>

          <tr>
            <td>
              <span className="label label-success">x1.25</span>
            </td>
            <td> {Time.speedWatch(seconds, 1.25)} </td>
          </tr>

          <tr>
            <td>
              <span className="label label-warning">x1.5</span>
            </td>
            <td> {Time.speedWatch(seconds, 1.5)} </td>
          </tr>

          <tr>
            <td>
              <span className="label label-danger">x2</span>
            </td>
            <td> {Time.speedWatch(seconds, 2)} </td>
          </tr>
        </tbody>
      </table>
    </div>
  );
}
