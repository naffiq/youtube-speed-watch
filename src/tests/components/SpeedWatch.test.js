import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SpeedWatch from './../../components/SpeedWatch';

describe('SpeedWatch', () => {
  it('should render properly', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SpeedWatch seconds={12} />, div);
  });

  it('should render proper time from seconds', () => {
    const component = renderer.create(
      <SpeedWatch seconds={1234} />
    );

    expect(component).toMatchSnapshot();
  });
});
