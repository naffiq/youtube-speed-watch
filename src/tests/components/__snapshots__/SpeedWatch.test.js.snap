exports[`SpeedWatch should render proper time from seconds 1`] = `
<div>
  <h2>
    Time to watch a video at different speeds:
  </h2>
  <table
    className="table">
    <tbody>
      <tr>
        <td>
          <span
            className="label label-default">
            x1
          </span>
        </td>
        <td>
           
          20:34
           
        </td>
      </tr>
      <tr>
        <td>
          <span
            className="label label-success">
            x1.25
          </span>
        </td>
        <td>
           
          16:27
           
        </td>
      </tr>
      <tr>
        <td>
          <span
            className="label label-warning">
            x1.5
          </span>
        </td>
        <td>
           
          13:43
           
        </td>
      </tr>
      <tr>
        <td>
          <span
            className="label label-danger">
            x2
          </span>
        </td>
        <td>
           
          10:17
           
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
